src/allmydata/test/test_connections.py

```diff
@@ -71,33 +72,42 @@ def test_launch_executable(self):
             self.assertEqual(f.mock_calls, [exp])
             self.assertIdentical(h, h1)
 
-    def test_socksport(self):
+    def test_socksport_unix_endpoint(self):
+        n = FakeNode(BASECONFIG+"[tor]\nsocks.port = unix:/var/lib/fw-daemon/tor_socks.socket\n")
+        h1 = mock.Mock()
+        with mock.patch("foolscap.connections.tor.socks_endpoint",
+                        return_value=h1) as f:
+            h = n._make_tor_handler()
+            self.assertTrue(IStreamClientEndpoint.providedBy(f.mock_calls[0]))
+            self.assertIdentical(h, h1)
+
+    def test_socks_endpoint(self):
         n = FakeNode(BASECONFIG+"[tor]\nsocks.port = tcp:127.0.0.1:1234\n")
         h1 = mock.Mock()
-        with mock.patch("foolscap.connections.tor.socks_port",
+        with mock.patch("foolscap.connections.tor.socks_endpoint",
                         return_value=h1) as f:
             h = n._make_tor_handler()
-            self.assertEqual(f.mock_calls, [mock.call("127.0.0.1", 1234)])
+            self.assertTrue(IStreamClientEndpoint.providedBy(f.mock_calls[0]))
             self.assertIdentical(h, h1)
 
-    def test_socksport_otherhost(self):
+    def test_socks_endpoint_otherhost(self):
         n = FakeNode(BASECONFIG+"[tor]\nsocks.port = tcp:otherhost:1234\n")
         h1 = mock.Mock()
-        with mock.patch("foolscap.connections.tor.socks_port",
+        with mock.patch("foolscap.connections.tor.socks_endpoint",
                         return_value=h1) as f:
             h = n._make_tor_handler()
-            self.assertEqual(f.mock_calls, [mock.call("otherhost", 1234)])
+            self.assertTrue(IStreamClientEndpoint.providedBy(f.mock_calls[0]))
             self.assertIdentical(h, h1)
 
     def test_socksport_bad_endpoint(self):
```

Comment:
oh, could you also change the names of these two tests (to test_socks_endpoint_bad_endpoint and test_socks_endpoint_not_integer), to match the change to test_socksport and test_socksport_otherhost ?